vimeo-networking/src/main/java/com/vimeo/networking/VimeoClient.java

```diff
@@ -940,20 +940,19 @@ public void failure(VimeoError error) {
             parameters.put(Vimeo.PARAMETER_VIDEO_DESCRIPTION, description);
         }
 
-        if (privacyValue != null) {
-            if (privacyValue == Privacy.PrivacyValue.PASSWORD) {
-                if (password == null) {
-                    callback.failure(new VimeoError("Password cannot be null password privacy type"));
+        if (privacySettingsParams != null && !privacySettingsParams.getParams().isEmpty()) {
+            final Map<String, Object> privacyMap = privacySettingsParams.getParams();
+            final PrivacyViewValue viewPrivacyValue = (PrivacyViewValue) privacyMap.get(Vimeo.PARAMETER_VIDEO_VIEW);
 
-                    return null;
+            if (viewPrivacyValue != null) {
```

Comment:
this null check is unnecessary, since the condition immediately following implicitly performs the same check